src/prefect/blocks/abstract.py

```diff
@@ -1,14 +1,40 @@
 from abc import ABC, abstractmethod
-from typing import Generic, TypeVar
+from typing import Generic, Optional, TypeVar
 
-from prefect import get_run_logger
 from prefect.blocks.core import Block
 from prefect.exceptions import MissingContextError
-from prefect.logging.loggers import get_logger
+from prefect.logging.loggers import get_logger, get_run_logger
 
 T = TypeVar("T")
 
 
+class NotificationBlock(Block, ABC):
+    """
+    A `Block` base class for sending notifications.
```

Comment:
We can take this opportunity to align the description of this abstract block type with the other abstract block type descriptions.
```suggestion
   Block that represents a resource in an external system that is able to send notifications.
```